test/Microsoft.AspNetCore.Razor.Evolution.Test/HtmlNodeOptimizationPassTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public class HtmlNodeOptimizationPassTest
+    {
+        [Fact]
+        public void Execute_CollapsesConditionalAttributes()
+        {
+            // Assert
+            var content = "<input value='hello world' />";
+            var sourceDocument = TestRazorSourceDocument.Create(content);
+            var originalTree = RazorSyntaxTree.Parse(sourceDocument);
+            var pass = new HtmlNodeOptimizationPass();
+            var codeDocument = RazorCodeDocument.Create(sourceDocument);
```

Comment:
You can use `TestRazorCodeDocument.Create(...)` here and it will accept the content directly